src/ripple/overlay/impl/PeerImp.cpp

```diff
@@ -86,6 +86,7 @@ PeerImp::PeerImp (Application& app, id_t id,
     , slot_ (slot)
     , request_(std::move(request))
     , headers_(request_)
+    , compressionEnabled_(headers_["X-Offer-Compression"] == "lz4" ? Compressed::On : Compressed::Off)
```

Comment:
Does this need `&& app_.config().COMPRESSION` as was done in the other `PeerImp` constructor?

Reply:
This is constructor for a peer making outbound connection. The header is not going to have the `X-Offer-Compression` header if the compression is not enabled so no need to check for `COMPRESSION`. The other constructor is for the inbound peer. We are not going to send compressed message to that peer if the compression is not enabled so we check for `COMPRESSION`.